### PR TITLE
UI: Refactor tool actions form

### PR DESCRIPTION
**Pull request by hellobontempo**
- [x] enterprise tests

Waiting until after the PR review to do the following (diff is much easier to read prior to these changes)
- [x] rename components and move to `tools/` folder 
- [x] co-locate .js and .hbs files

### CODE REVIEW

File: ui/app/templates/components/tool-actions-form.hbs
Comment (hellobontempo) [inline ui/app/templates/components/tool-actions-form.hbs:1]: These components now render directly in `vault/cluster/tools.hbs`

File: ui/app/components/tool-hash.js [hunk 1]
```diff
@@ -5,29 +5,57 @@
...
+    this.sum = null;
+    this.errorMessage = '';
+  }
+
   @action
-  onClear() {
-    this.args.onClear();
+  handleEvent(evt) {
+    const { name, value } = evt.target;
+    this[name] = value;
```
Comment (Monkeychip) [inline ui/app/components/tool-hash.js:41]: neat solution

File: ui/app/components/tool-random.js [hunk 2]
```diff
@@ -4,31 +4,52 @@
...
   @action
-  onClear() {
-    this.args.onClear();
+  handleEvent(evt) {
+    const { name, value } = evt.target;
+    this[name] = value;
+  }
+
+  @action
+  async handleSubmit(evt) {
```
Comment (Monkeychip) [inline ui/app/components/tool-random.js:43]: do we ever need to reset the error message to empty here?

**Review by Monkeychip — APPROVED** [review]
Really great work on this!

File: ui/app/components/tool-random.js (hunk 2 above)
Comment (hellobontempo) [reply ui/app/components/tool-random.js:43]: Good question! If there are two unsuccessful requests, the error message will update accordingly. On success, the view changes from the form to the "show" view and all of the tracked variables clear on `Done`(when a user is finished with the show page) so if there is an error but the subsequent request is successful the error message does not still render. So after thinking through the two scenarios, I don't think we need to clear it in this action 🤔

**Pull request merged**